daemon/main.cpp

```diff
@@ -273,27 +299,20 @@ main(int argc, char *argv[])
     return 0;
   }
 
-  // this is important, can downgrade from Info though
-  llarp::LogInfo("Running from: ", fs::current_path().string());
-  llarp::LogInfo("Using config file: ", conffname);
-  ctx      = llarp_main_init(conffname.c_str(), multiThreaded);
-  int code = 1;
-  if(ctx)
+  std::thread main_thread{
+      std::bind(&run_main_context, conffname, multiThreaded, debugMode)};
+  auto ftr = exit_code.get_future();
+  do
   {
-    signal(SIGINT, handle_signal);
-    signal(SIGTERM, handle_signal);
-#ifndef _WIN32
-    signal(SIGHUP, handle_signal);
-#endif
+    // do periodic non lokinet related tasks here
+  } while(ftr.wait_for(std::chrono::seconds(1)) != std::future_status::ready);
```

Comment:
Would be nice to take this period as a command line argument, but can be a future extension point.